central/cve/csv/handler.go

```diff
@@ -118,92 +126,204 @@ func CVECSVHandler() http.HandlerFunc {
+			dataRow.DeploymentCount = fmt.Sprint(deploymentCount)
 			// Entity counts should be scoped to CVE only
 			imageCount, err := d.ImageCount(ctx, resolvers.RawQuery{})
 			if err != nil {
 				errorList.AddError(err)
 			}
-			dataRow.imageCount = fmt.Sprint(imageCount)
+			dataRow.ImageCount = fmt.Sprint(imageCount)
 			// Entity counts should be scoped to CVE only
 			nodeCount, err := d.NodeCount(ctx, resolvers.RawQuery{})
 			if err != nil {
 				errorList.AddError(err)
 			}
-			dataRow.nodeCount = fmt.Sprint(nodeCount)
+			dataRow.NodeCount = fmt.Sprint(nodeCount)
 			// Entity counts should be scoped to CVE only
 			componentCount, err := d.ComponentCount(ctx, resolvers.RawQuery{})
 			if err != nil {
 				errorList.AddError(err)
 			}
-			dataRow.componentCount = fmt.Sprint(componentCount)
+			dataRow.ComponentCount = fmt.Sprint(componentCount)
 			scannedTime, err := d.LastScanned(ctx)
 			if err != nil {
 				errorList.AddError(err)
 			}
-			dataRow.scannedTime = csv.FromGraphQLTime(scannedTime)
+			dataRow.ScannedTime = csv.FromGraphQLTime(scannedTime)
 			publishedTime, err := d.PublishedOn(ctx)
 			if err != nil {
 				errorList.AddError(err)
 			}
-			dataRow.publishedTime = csv.FromGraphQLTime(publishedTime)
-			dataRow.summary = d.Summary(ctx)
-
-			output.addRow(dataRow)
+			dataRow.PublishedTime = csv.FromGraphQLTime(publishedTime)
+			dataRow.Summary = d.Summary(ctx)
 
+			cveRows = append(cveRows, dataRow)
 			if err := errorList.ToError(); err != nil {
-				log.Errorf("failed to generate complete csv entry for cve %s: %v", dataRow.cveID, err)
+				log.Errorf("failed to generate complete csv entry for cve %s: %v", dataRow.CVE, err)
 			}
 		}
+		return cveRows, nil
+	}
 
-		output.Write(w, "cve_export")
+	cveType := search.GetFieldValueFromQuery(rawQuery.String(), search.CVEType)
+	switch cveType {
+	case "":
+		return nil, errors.New("'CVE Type' filter required but not found in input query")
+	case storage.CVE_IMAGE_CVE.String():
+		imageCveRows, err := imageCveCsv.ImageCVECSVRows(c, query, rawQuery, paginatedQuery)
+		if err != nil {
+			return nil, err
+		}
+		return imageCVERowsToCVERows(imageCveRows), nil
+	case storage.CVE_NODE_CVE.String():
+		nodeCveRows, err := nodeCveCsv.NodeCVECSVRows(c, query, rawQuery, paginatedQuery)
+		if err != nil {
+			return nil, err
+		}
+		return nodeCVERowsToCVERows(nodeCveRows), nil
+	case storage.CVE_K8S_CVE.String(), storage.CVE_OPENSHIFT_CVE.String(), storage.CVE_ISTIO_CVE.String():
+		clusterCveRows, err := clusterCveCsv.ClusterCVECSVRows(c, query, rawQuery, paginatedQuery)
+		if err != nil {
+			return nil, err
+		}
+		return clusterCVERowsToCVERows(clusterCveRows), nil
+	default:
+		return nil, errors.Errorf("Unexpected value for 'CVE Type' filter. Value should be one of '%s', '%s', '%s', '%s', '%s'",
+			storage.CVE_IMAGE_CVE.String(), storage.CVE_NODE_CVE.String(), storage.CVE_K8S_CVE.String(), storage.CVE_OPENSHIFT_CVE.String(), storage.CVE_ISTIO_CVE.String())
```

Comment:
Move this above. The change is not subject to the feature flag. API expects cve type to be provided starting 3.73.0. 

```
- Starting 3.73, CSV export API `/api/vm/export/csv` would require to pass `CVE Type` filter as part of the input query parameter. Requests that do not have the filter would error out.
  - Examples : `CVE Type:NODE_CVE`, `CVE_Type:IMAGE_CVE`, `CVE_TYPE:K8S_CVE`
```